### PR TITLE
 Implement dotenv.

### PR DESCRIPTION
**Pull request by ghost**
This PR adds dotenv to the application.

`dotenv` will use the `.env` file to feed the `process.env` object in a transparent manner.

Also, this approach follows the 12factor principles:

> The twelve-factor app stores config in environment variables (often shortened to env vars or env). Env vars are easy to change between deploys without changing any code; unlike config files, there is little chance of them being checked into the code repo accidentally

### CODE REVIEW

File: data/default-config.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "channel": "legi-bot",
+  "channel": "legisacn",
```
Comment (finnbear) [inline data/default-config.json]: typo

File: .env.template [hunk 2]
```diff
@@ -0,0 +1,2 @@
+DISCORD_CLIENT_KEY = <your-client-key>
```
Comment (finnbear) [inline .env.template]: Client token, not client key

**Review by finnbear — CHANGES_REQUESTED** [review]
See comments :)

Also, explain why this is better than the JSON approach.

**Comment by ghost** [comment]
@finnbear here you go.

File: .gitignore [hunk 3]
```diff
@@ -28,14 +28,16 @@ yarn.lock
...
+
 # Ignore cache and temporary folders.
 .cache
 .temp
 
 # Ignore Junit reporters results.
 junit.xml
 
-# Ignore some application files (such as credentials).
-credentials.json
```
Comment (finnbear) [inline .gitignore:39]: keep in gitignore or ensure deletion

**Pull request merged**